### PR TITLE
Parse JToken correctly for empty and null string in host.json

### LINKED ISSUE #4332: Check for null when parsing functionTimeout from host.json
#### Investigative information
Same issue: https://github.com/Azure/azure-functions-host/issues/2468

Please provide the following:

- Timestamp: 04-18-2019 1000
- Function App version (1.0 or 2.0): 2x
- Function App name: Any
- Function name(s) (as appropriate): Any
- Invocation ID:
- Region:

#### Repro steps

Provide the steps required to reproduce the problem:

Set below attribute in host.json
"functionTimeout": null,

#### Expected behavior

Provide a description of the expected behavior.

Based on code, it should fail at  "if (value != null)" condition and set default timeout.
Based on 1x behavior, have unlimited timeout.
Documentation should be clear on the behavior.  Seems it was not addressed in https://github.com/Azure/azure-functions-host/issues/3603

#### Actual behavior

Provide a description of the actual behavior observed.

Gives start error: 
String was not recognized as a valid TimeSpan.

#### Known workarounds

Provide a description of any known workarounds.

Avoid setting null

### PR DESCRIPTION
**Pull request by jijohn14**
When host.json has setting as "functionTimeout": null, it is translated to empty string as value. Fixing this.
Fixes #4332

### CODE REVIEW

File: src/WebJobs.Script/Config/HostJsonFileConfigurationSource.cs [hunk 1]
```diff
@@ -102,7 +102,7 @@ private void ProcessToken(JToken token)
                     case JTokenType.Bytes:
                     case JTokenType.TimeSpan:
                         string key = ConfigurationSectionNames.JobHost + ConfigurationPath.KeyDelimiter + ConfigurationPath.Combine(_path.Reverse());
-                        Data[key] = token.Value<JValue>().ToString(CultureInfo.InvariantCulture);
+                        Data[key] = token.Value<string>()?.ToString(CultureInfo.InvariantCulture);
```
Comment (pragnagopa) [inline src/WebJobs.Script/Config/HostJsonFileConfigurationSource.cs]: The right fix was in the earlier PR. We do not want parse other token types as string. This change makes sense for token string.

File: src/WebJobs.Script/Config/HostJsonFileConfigurationSource.cs [hunk 2]
```diff
@@ -102,7 +102,7 @@ private void ProcessToken(JToken token)
                     case JTokenType.Bytes:
                     case JTokenType.TimeSpan:
```
Comment (pragnagopa) [inline src/WebJobs.Script/Config/HostJsonFileConfigurationSource.cs]: right fix was in the earlier PR. In ScriptHostOptionsSetup.cs 

```
private void ConfigureFunctionTimeout(IConfigurationSection jobHostSection, ScriptJobHostOptions options)
        {
            if (options.FunctionTimeout != null)
            {
                ValidateTimeoutValue(options, options.FunctionTimeout);
            }
            else
            {
                TimeSpan functionTimeout = _environment.IsDynamic() ? DefaultFunctionTimeoutDynamic : DefaultFunctionTimeout;
                options.FunctionTimeout = functionTimeout;
            }
        }
```
Options is already binding to functiontimeout as timespan. There is no need to convert it back to string.

File: src/WebJobs.Script/Config/HostJsonFileConfigurationSource.cs (hunk 1 above)
Comment (jitinjohn) [reply src/WebJobs.Script/Config/HostJsonFileConfigurationSource.cs]: Existing code is converting the token to string using ToString(). Difference is that previous change converts functionTimeout: null to string.empty while this keeps it null

Comment (pragnagopa) [reply src/WebJobs.Script/Config/HostJsonFileConfigurationSource.cs]: JValue is the right type we need to convert to. Null is a right type. The issue is in the way code was parsing the fucntiontimeout specifically

**Comment by pragnagopa** [comment]
Closing in favor  of https://github.com/Azure/azure-functions-host/pull/4361

**Comment by jitinjohn** [comment]
Abandoning this in favor of #4361

**Pull request closed without merging**